src/query/v2/requests.hpp

```diff
@@ -0,0 +1,201 @@
+  std::vector<std::string> property_names;
+  std::vector<Expression> expressions;
+  bool only_unique = false;
+  std::optional<std::vector<OrderBy>> order_by;
+  std::optional<size_t> limit;
+  std::optional<Filter> filter
+};
+
+struct GetPropertiesResponse {
+  Values values;
+  std::optional<std::unordered_map<size_t, std::string>> property_name_map;
+};
+
+enum EdgeDirection { OUT = 1; IN = 2; BOTH = 3; };
+
+struct ExpandOneRequest {
+  Hlc transaction_id;
+  std::vector<VertexId> src_vertices;
+  std::vector<EdgeType> edge_types;
+  EdgeDirection direction;
+  bool only_unique_neighbor_rows = false;
+  //  The empty optional means return all of the properties, while an empty
+  //  list means do not return any properties
+  //  TODO(antaljanosbenjamin): All of the special values should be communicated through a single vertex object
+  //                            after schema is implemented
+  //  Special values are accepted:
+  //  * __mg__labels
+  std::optional<std::vector<std::string>> src_vertex_properties;
+  //  TODO(antaljanosbenjamin): All of the special values should be communicated through a single vertex object
+  //                            after schema is implemented
+  //  Special values are accepted:
+  //  * __mg__dst_id (Vertex, but without labels)
+  //  * __mg__type (binary)
+  std::optional<std::vector<std::string>> edge_properties;
+  //  QUESTION(antaljanosbenjamin): Maybe also add possibility to expressions evaluated on the source vertex?
+  //  List of expressions evaluated on edges
+  std::vector<Expression> expressions;
+  std::optional<std::vector<OrderBy>> order_by;
+  std::optional<size_t> limit;
+  std::optional<Fitler> filter;
+};
+
+struct ExpandOneResultRow {
+  // NOTE: This struct could be a single Values with columns something like this:
+  // src_vertex(Vertex), vertex_prop1(Value), vertex_prop2(Value), edges(list<Value>)
+  // where edges might be a list of:
+  // 1. list<Value> if only a defined list of edge properties are returned
+  // 2. map<binary, Value> if all of the edge properties are returned
+  // The drawback of this is currently the key of the map is always interpreted as a string in Value, not as an
+  // integer, which should be in case of mapped properties.
+  Vertex src_vertex;
+  std::optional<Values> src_vertex_properties;
+  Values edges;
+};
+
+struct ExpandOneResponse {
+  // This approach might not suit the expand with per shard parrallelization,
+  // because the property_name_map has to be accessed from multiple threads
+  // in order to avoid duplicated keys (two threads might map the same
+  // property with different numbers) and multiple passes (to unify the
+  // mapping amond result returned from different shards).
+  std::vector<ExpandOneResultRow> result;
+  std::optional<std::unordered_map<size_t, std::string>> property_name_map;
+};
+
+struct NewVertex {
+  std::vector<size_t> label_ids;
+  std::map<size_t, Value> properties;
```

Comment:
should this be PropertyId instead of size_t?

Reply:
yes and the above vector should be `std::vector<Label>` which is basically the label_id